pkg/ccl/logictestccl/testdata/logic_test/partitioning_deinterleave

```diff
@@ -0,0 +1,489 @@
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE parent]
+----
+CREATE TABLE public.parent (
+  i INT8 NOT NULL,
+  CONSTRAINT "primary" PRIMARY KEY (i ASC),
+  FAMILY "primary" (i)
+) PARTITION BY LIST (i) (
+  PARTITION negative VALUES IN ((DEFAULT), (-1)),
+  PARTITION zero VALUES IN ((0)),
+  PARTITION positive VALUES IN ((1))
+);
+ALTER PARTITION negative OF INDEX test.public.parent@primary CONFIGURE ZONE USING
+  constraints = '[+region=ap-southeast-2]';
+ALTER PARTITION positive OF INDEX test.public.parent@primary CONFIGURE ZONE USING
+  constraints = '[+region=us-east-1]';
+ALTER PARTITION zero OF INDEX test.public.parent@primary CONFIGURE ZONE USING
+  constraints = '[+region=ca-central-1]'
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE child]
+----
+CREATE TABLE public.child (
+  i INT8 NOT NULL,
+  j INT8 NOT NULL,
+  CONSTRAINT "primary" PRIMARY KEY (i ASC, j ASC),
+  INDEX idx_j_i (j ASC, i ASC) PARTITION BY LIST (j) (
+    PARTITION negative VALUES IN ((DEFAULT), (-1)),
+    PARTITION zero VALUES IN ((0)),
+    PARTITION positive VALUES IN ((1))
+  ),
+  FAMILY "primary" (i, j)
+) INTERLEAVE IN PARENT public.parent (i)
+-- Warning: Partitioned table with no zone configurations.
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE grandchild]
+----
+CREATE TABLE public.grandchild (
+  i INT8 NOT NULL,
+  j INT8 NOT NULL,
+  k INT8 NOT NULL,
+  CONSTRAINT "primary" PRIMARY KEY (i ASC, j ASC, k ASC),
+  INDEX idx_k_j_i (k ASC, j ASC, i ASC) PARTITION BY LIST (k) (
+    PARTITION negative VALUES IN ((DEFAULT), (-1)),
+    PARTITION zero VALUES IN ((0)),
+    PARTITION positive VALUES IN ((1))
+  ),
+  FAMILY "primary" (i, j, k)
+) INTERLEAVE IN PARENT public.child (i, j);
+ALTER PARTITION negative OF INDEX test.public.grandchild@idx_k_j_i CONFIGURE ZONE USING
+  constraints = '[+region=ap-southeast-2]';
+ALTER PARTITION positive OF INDEX test.public.grandchild@idx_k_j_i CONFIGURE ZONE USING
+  constraints = '[+region=us-east-1]';
+ALTER PARTITION zero OF INDEX test.public.grandchild@idx_k_j_i CONFIGURE ZONE USING
+  constraints = '[+region=ca-central-1]'
+
+statement error cannot change primary key of table child because table\(s\) \[grandchild\] are interleaved into it
+ALTER TABLE child ALTER PRIMARY KEY USING COLUMNS (i);
+
+statement ok
+ALTER TABLE grandchild ALTER PRIMARY KEY USING COLUMNS (i);
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE grandchild]
+----
+CREATE TABLE public.grandchild (
+  i INT8 NOT NULL,
+  j INT8 NOT NULL,
+  k INT8 NOT NULL,
+  CONSTRAINT "primary" PRIMARY KEY (i ASC),
+  UNIQUE INDEX grandchild_i_j_k_key (i ASC, j ASC, k ASC),
+  INDEX idx_k_j_i (k ASC, j ASC, i ASC) PARTITION BY LIST (k) (
+    PARTITION negative VALUES IN ((DEFAULT), (-1)),
+    PARTITION zero VALUES IN ((0)),
+    PARTITION positive VALUES IN ((1))
+  ),
+  FAMILY "primary" (i, j, k)
+);
```

Comment:
am i missing something or is a PK partitioning missing here?

Reply:
🤦 fixed.